### PR TITLE
update pop merkle tree

### PR DESCRIPTION
**Pull request by Mr-Leshiy**

### CODE REVIEW

File: src/vbk/merkle.hpp [hunk 1]
```diff
@@ -22,6 +24,23 @@
...
+template <typename pop_t>
+void popDataToHash(const std::vector<pop_t>& data, std::vector<uint256>& leaves)
+{
+    for (const auto& el : data) {
+        auto id = el.getId();
+        std::vector<uint8_t> hash(32, 0);
+        for (size_t i = 0; i < hash.size() && i < id.size(); ++i) {
+            hash[i] = id[i];
+        }
+        leaves.push_back(uint256(hash));
```
Comment (Warchant) [inline src/vbk/merkle.hpp]:
```suggestion
        leaves.push_back(uint256(el.getId().asVector()));
```

File: src/vbk/merkle.hpp [hunk 2]
```diff
@@ -39,17 +58,18 @@ inline int GetPopMerkleRootCommitmentIndex(const CBlock& block)
 inline uint256 BlockPopDataMerkleRoot(const CBlock& block)
 {
     std::vector<uint256> leaves;
-    auto bytes = block.popData.toVbkEncoding();
-    leaves.push_back(Hash(bytes.begin(), bytes.end()));
+
+    popDataToHash(block.popData.context, leaves);
+    popDataToHash(block.popData.atvs, leaves);
+    popDataToHash(block.popData.vtbs, leaves);
```
Comment (Warchant) [inline src/vbk/merkle.hpp]: context, then VTBs, then ATVs

File: src/vbk/merkle.hpp (hunk 1 above)
Comment (Warchant) [reply src/vbk/merkle.hpp]:
```
uint256 leaf;
std::copy(id.begin(), id.end(), leaf.begin();

```

File: src/vbk/merkle.hpp (hunk 2 above)
Comment (Mr-Leshiy) [reply src/vbk/merkle.hpp]: Fixed

File: src/vbk/merkle.hpp (hunk 1 above)
Comment (Mr-Leshiy) [reply src/vbk/merkle.hpp]: Fixed

**Comment by Warchant** [comment]
Will be merged on next rebootstrap/restart of the network

**Review by Warchant — APPROVED** [review]

**Pull request merged**